### PR TITLE
Telemetry module for Swift

### PR DESCRIPTION
**Pull request by MichaelGHSeg**

### CODE REVIEW

File: Sources/Segment/Utilities/Telemetry.swift [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    private var queueBytes = 0
+    private var queueSizeExceeded = false
+    private var seenErrors = [String: Int]()
+    public var started = false
+    private var rateLimitEndTime: TimeInterval = 0
+    private var telemetryQueue = DispatchQueue(label: "telemetryQueue")
+    private var telemetryTimer: Timer?
+
+    func start() {
+        guard enable, !started, sampleRate != 0.0 else { return }
```
Comment (didiergarcia) [inline Sources/Segment/Utilities/Telemetry.swift]: Maybe `sampleRate > 0.0 && sampleRate <= 1.0` ?

File: Sources/Segment/Utilities/Telemetry.swift [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+        return request
+    }
+
+    private func queueHasSpace() -> Bool {
+        var over = false
+        telemetryQueue.sync {
+            over = queue.count < maxQueueSize
+        }
+        return over
```
Comment (didiergarcia) [inline Sources/Segment/Utilities/Telemetry.swift]: I think value determined is if you are "under" max queue size. Should that variable be changed to `under` ?

**Review by didiergarcia — APPROVED** [review]
Looks good overall.

I would add some better protection to the sample rate by given it a bounds like > 0.0 and <= 1.0 if that's the intension.

File: Sources/Segment/Utilities/Telemetry.swift (hunk 2 above)
Comment (MichaelGHSeg) [reply Sources/Segment/Utilities/Telemetry.swift]: That would make more sense.  I like the guard clause but the inverted logic got me a bit.

**Pull request merged**